cmd/disk-cache.go

```diff
@@ -846,7 +846,7 @@ func (c cacheObjects) CompleteMultipartUpload(ctx context.Context, bucket, objec
 		return
 	}
 	// create new multipart upload in cache with same uploadID
-	dcache.CompleteMultipartUpload(ctx, bucket, object, uploadID, uploadedParts, opts)
+	dcache.completeMultipartUpload(ctx, bucket, object, uploadID, objInfo.ETag, uploadedParts, opts)
 	return
```

Comment:
Not sure if I am understanding the PR correctly, but isnt the better approach to simply call dcache.completeMultipartUpload like previously and only write a little code to over-write the ETag coming from the backend into the cache backend?

Reply:
issue is Part ETag for gateway backends need not be md5sum - fsObjects CompleteMultipartUpload implementation generates the ETag for finalized upload by creating md5hash of concatenated part ETags (see getCompleteMultipartMD5 in object-api-utils.go). If we rely on fsObjects implementation as in master, the part ETag cannot be hex decoded causing overall operation to fail. For cache purposes, we just need to persist the backend ETag in the cache - hence the need for this change.

Reply:
Yeah i meant that we could add some code to only overwrite the ETag in the cache fs backend alone. This way we can inherit any changes to the CompleteMultipartUpload code - otherwise doing a reimplementation like this will cause some overhead to make sure the two implementations stay in sync.